mbed/mbed.py

```diff
@@ -1236,6 +1236,9 @@ def clone(self, url, path, rev=None, depth=None, protocol=None, offline=False, *
                     continue
                 try:
                     scm.clone(url, path, depth=depth, protocol=protocol, **kwargs)
+                    if scm.name == 'bld':
+                        with cd(path):
+                            scm.update()
```

Comment:
Does this need to happen only for `bld` or is this also valid for other scms?

Reply:
It's a bld specific thing. 

With real SCMs (Git/Hg) you want to get as much as possible from the index from the clone() operation as you can then freely change between tags. Bld is emulating SCM though, we don't want anything checked out during Bld.clone() with no revision, because this will deliver a 300MB codebase. Bld.clone() rather "initializes" a build SCM, and then needs an update() to either latest or a specific revision.

Reply:
Alright. I would like to revisit this, as it seems like we're breaking the SCM abstraction here.

Reply:
@theotherjimmy I actually rewrote so we're not breaking the SCM abstraction. Check what was merged in.

Reply:
Yeah, I noticed. That's why I merged!